server/drafty/drafty.go

```diff
@@ -37,7 +93,137 @@ var tags = map[string]spanfmt{
 	"EX": {"", true},
 }
 
-// ToPlainText converts message payload from Drafy format to string.
+// Preview shortens Drafty to the specified length (in runes) and removes large content from entities making them
+// suitable for a one-line preview, for example for showing in push notifications.
+func Preview(content interface{}, length int) (interface{}, error) {
+	if content == nil {
+		return "", nil
+	}
+
+	var original map[string]interface{}
+
+	switch tmp := content.(type) {
+	case string:
+		return tmp, nil
+	case map[string]interface{}:
+		original = tmp
+	default:
+		return "", errUnrecognizedContent
+	}
+
+	txt, txtOK := original["txt"].(string)
+	fmt, fmtOK := original["fmt"].([]interface{})
+	ent, entOK := original["ent"].([]interface{})
+
+	// At least one must be set.
+	if !txtOK && !fmtOK && !entOK {
+		return "", errUnrecognizedContent
+	}
+
+	var textLen int
+	preview := make(map[string]interface{})
+	if txtOK {
+		runes := []rune(txt)
+		textLen = len(runes)
+		if textLen > length {
+			preview["txt"] = string(runes[:length])
+			textLen = length
+		} else {
+			preview["txt"] = txt
+		}
+		txt = preview["txt"].(string)
```

Comment:
In case the original `textLen` is less than `length` (the above else clause), this assignment is also not necessary.
That said, `preview["txt"]` needs to be modified only when the original `textLen` exceeds `length`.

Reply:
Done